enhancements/support-for-realtime-kernel.md

```diff
@@ -0,0 +1,292 @@
+kernel will also update the `kernel-rt` packages if updates are available in
+the `machine-os-content` image.  This treats the `kernel-rt` package like any
+other RPM installed on the RHCOS nodes.
+
+### Test Plan
+
+- Test install of `kernel-rt` packages on single RHCOS node
+- Test removal of `kernel-rt` packages on single RHCOS node
+- Test `kernelType: realtime` on OCP cluster with RHCOS nodes using default kernel
+- Test `kernelType: default` on OCP cluster with RHCOS nodes using default kernel
+- Test `kernelType: default` on OCP cluster with RHCOS nodes already using real-time kernel
+- Test `kernelType: realtime` on OCP cluster with RHEL nodes
+- Test `kernelType: default` on OCP cluster with RHEL nodes
+- Test upgrade of OCP cluster with RHCOS nodes using real-time kernel (no `kernel-rt` updates in `machine-os-content`)
+- Test upgrade of OCP cluster with RHCOS nodes using real-time kernel (`kernel-rt` updates available in `machine-os-content`)
+- Test `kernelType: realtime` on older OCP cluster with no real-time kernel support
+
+### Graduation Criteria
+
+For this feature to be considered stable:
+- `kernel-rt` packages available in `machine-os-content` images
+- MachineConfig spec has `kernelType` field
+- Cluster with RHCOS nodes can successfully select + use the `kernel-rt` packages
+- Cluster with RHCOS nodes using `kernel-rt` can be upgraded successfully
+
+### Upgrade / Downgrade Strategy
+
+- Nodes using `kernel-rt` packages should continue to use `kernel-rt` packages through upgrades
+- Nodes using `kernel-rt` packages should continue to use `kernel-rt` packages after a downgrade.
+    - **WARNING**: downgrading a cluster to a previous version that does not have real-time kernel support is unsupported.
+
+### Version Skew Strategy
+
+There should be no implications for real-time kernel support if other components in the cluster are at a different version.
+
+## Implementation History
+
+- 2019-12-11: RT kernel included in `machine-os-content` image - https://url.corp.redhat.com/rhcos-rt-kernel
+- 2019-12-12: PR for MCO support - https://github.com/openshift/machine-config-operator/pull/1330
+
+## Drawbacks
+
+- If there are additional changes required to cluster components for full
+support of the real-time kernel, this proposal has not been scoped for that
+and should be pushed to a later release.
+
+- It should be noted that this increases the support matrix for the
+OpenShift support folks.  The amount of additional configurations that
+this could create (i.e. disk encryption on/off, different platforms,  etc.)
+is something that needs to be considered when reviewing the proposal.
+
+- Customers requiring use of custom kernel modules (or [SRO](https://github.com/zvonkok/special-resource-operator))
+for things like nVidia GPUs will need to rebuild the kernel modules
+to support the real-time kernel.
+
+- Customers requiring FIPS support **SHOULD NOT** use the real-time kernel.
```

Comment:
Can we handle this situation explicitly in the MCO? Like, the MCO can refuse to enable it (e.g. by setting the kerneltype=default on the rendered MC when fips=true)?

Reply:
Seems feasible.  @sinnykumari WDYT?

Reply:
yes should be possible. MCO can print some message saying Not supported when FIPS is already enabled.